backend/src/main/java/com/beaplab/BeaplabEngine/controller/WatchController.java

```diff
@@ -0,0 +1,336 @@
+
+        if (downloadedFile != null) {
+
+            jsonObject.put(BeapEngineConstants.SUCCESS_STR, true);
+            jsonObject.put("status_code", HttpStatus.OK.value());
+            jsonObject.put("file", downloadedFile);
+            return new ResponseEntity<>(jsonObject, HttpStatus.OK);
+
+        } else {
+            rollbar.info("Requested .zip file not found at the server");
+            jsonObject.put(BeapEngineConstants.SUCCESS_STR, false);
+            jsonObject.put("message", "Requested .zip file not found at the server");
+            jsonObject.put("status_code", HttpStatus.NOT_FOUND.value());
+            return new ResponseEntity<>(jsonObject, HttpStatus.NOT_FOUND);
+        }
+    }
+
+
+    @RequestMapping(value = "/{watchType}/list/{type}", method = RequestMethod.GET)
+    @ApiOperation(value = "Apple Watch data list", notes = "List the user's Apple Watch data.")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "{success: true, predicted_id: XXXX}"),
+            @ApiResponse(code = 406, message = "{success: false, message: Invalid predictionModel}"),
+            @ApiResponse(code = 500, message = "{success: false, message: XYZ}")
+    })
+    public ResponseEntity<JSONObject> list(HttpServletRequest request,
+                                           @ApiParam(value = "type of the apple watch data to be listed: raw, processed, predicted", required = true)
+                                           @PathVariable("type") String type,
+                                           @ApiParam(value = "the watch type to be downloaded", required = true)
+                                           @PathVariable("watchType") String watchType
+                                           ) {
+
+        rollbar.info("in WatchController/list");
+
+        HttpSession session = request.getSession(false);
+        if(session == null || !request.isRequestedSessionIdValid()) { // invalid session
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put(BeapEngineConstants.SUCCESS_STR, false);
+            jsonObject.put("message", "Invalid session");
+            jsonObject.put("status_code", HttpStatus.NON_AUTHORITATIVE_INFORMATION.value());
+
+            return new ResponseEntity<>(jsonObject, (HttpStatus.valueOf((int)jsonObject.get("status_code"))));
+        }
+
+        SessionDetails sessionDetails = (SessionDetails) session.getAttribute("SESSION_DETAILS");
+
+        JSONObject result = new JSONObject();
+        switch (watchType) {
+            case "applewatch":
+                result = appleWatchService.list(sessionDetails.getUserId(), type);
+                break;
+            case "fitbit":
+                result = fitbitService.list(sessionDetails.getUserId(), type);
+                break;
+            default:
+                break;
+        }
+        
+        return new ResponseEntity<>(result, (HttpStatus.valueOf((int)result.get("status_code"))));
+    }
+
+}
```

Comment:
This should just work, right? Like the API routes should automatically link?

Reply:
yep its all automatically handled